src/GraphQL.Analyzers/NotAGraphTypeAnalyzer.cs

```diff
@@ -85,10 +84,10 @@ private void AnalyzeGenericName(SyntaxNodeAnalysisContext context)
 
         bool HasNotAGraphTypeAttribute(ISymbol typeParam)
         {
-            notAGraphTypeAttribute ??= context.Compilation.GetTypeByMetadataName(Constants.MetadataNames.NotAGraphTypeAttribute);
             return typeParam.GetAttributes()
-                .Any(data => SymbolEqualityComparer.Default.Equals(data.AttributeClass, notAGraphTypeAttribute));
+                .Any(data => data.AttributeClass?.MetadataName == Constants.MetadataNames.NotAGraphTypeAttribute);
```

Comment:
`GetTypeByMetadataName` returns `null` when the same name is defined in multiple referenced assemblies. Since code analysis attributes are now added as linked sources to multiple projects, this approach doesn't work anymore.
There is another method `GetAllTypesByMetadataName` that will return all the types with matching metadata name but it's not available in the used version.

Reply:
I've not read through the code, so I'm not sure exactly how this affects usage.